application/encoding.go

```diff
@@ -77,8 +77,12 @@ func UnmarshalResponse(t int, msg []byte) *protocol.Response {
 			Error: res.Error,
 		}
 		if err := response.Validate(); err != nil {
-			return &protocol.Response{
-				Error: protocol.ErrMalformedMessage,
+			// we don't want to return an ErrMalformedMessage
```

Comment:
I think this is covered in https://github.com/coniks-sys/coniks-go/blob/master/protocol/message.go#L272

Reply:
Yes, but the problem is that *this* function was still returning an `ErrMalformedMessage` even if the error is in `errors`. In other words, because `Validate()` in message.go returns `msg.Error`, which gives `err == nil` after `Validate()` returns, the `return &protocol.Response{Error: protocol.ErrMalformedMessage }` statement was always being returned, even when the `msg.Error` was in `errors`.

Reply:
I see. I gave it a try in https://github.com/coniks-sys/coniks-go/compare/unmarshalling. If it's ok, feel free to merge to this branch.

Reply:
Those changes looks good to me. Thank you! I also like that you re-wrote the test cases. I'll merge this branch when I get a chance.